packages/flutter/test/material/refresh_indicator_test.dart

```diff
@@ -406,15 +404,31 @@ void main() {
       ),
     );
 
+    // test cancel
+
+    await tester.drag(find.text('A'), const Offset(0.0, kExpectedArmOffset - 1));
+    await tester.pump();
+    final Offset position = tester.getTopLeft(find.byType(RefreshProgressIndicator));
+    // wait before it starts moving
+    await tester.pump(const Duration(milliseconds: 1));
+    await tester.pump(const Duration(milliseconds: 1));
+    expect(tester.getTopLeft(find.byType(RefreshProgressIndicator)).dy, lessThan(position.dy));
+    expect(refreshCalled, false);
+    await tester.pumpAndSettle();
+
+    // test refresh
+
     await tester.fling(find.text('A'), const Offset(0.0, 300.0), 1000.0);
     await tester.pump(const Duration(milliseconds: 100));
     expect(lastScrollOffset = controller.offset, lessThan(0.0));
-    expect(refreshCalled, isFalse);
+    expect(refreshCalled, false);
 
-    await tester.pump(const Duration(milliseconds: 100));
+    await tester.pump(const Duration(milliseconds: 1));   // accept refresh
+    await tester.pump(const Duration(milliseconds: 150)); // wait for animation
+    await tester.pump(const Duration(milliseconds: 1));   // wait for the refresh callback to be called
```

Comment:
This test was tracking an incorrect behavior previously - instead of animating to refresh position, it was moving along with overscroll for one frame. Without clamping on [this line](https://github.com/flutter/flutter/blob/d290763941e3d71ad77907ac5fcc88032bb5d06d/packages/flutter/lib/src/material/refresh_indicator.dart#L398), which was removed to make the cancellation possible, the indicator created an animation, and the test failed, without these 2 additional pumps.